code/modules/antagonists/separatist/nation_creation.dm

```diff
@@ -0,0 +1,72 @@
+
+/**
+ * ### create_separatist_nation()
+ *
+ * Helper called to create the separatist antagonist via making a department independent from the station.
+ *
+ * * Arguments:
+ * * department: which department to revolt. if null, will pick a random non-independent department. starts as a type, then turns into the reference to the singleton.
+ * * announcement: whether to tell the station a department has gone independent.
+ * * dangerous: whether this nation will have objectives to attack other independent departments, requires more than one nation to exist obviously
+ *
+ * Returns null if everything went well, otherwise a string describing what went wrong.
+ */
+/proc/create_separatist_nation(datum/job_department/department, announcement = FALSE, dangerous = FALSE)
+
+	var/list/jobs_to_revolt = list()
+	var/list/citizens = list()
+
+	//departments that are already independent, these will be disallowed to be randomly picked
+	var/list/independent_departments = list()
+	//reference to all independent nation teams
+	var/list/team_datums = list()
+	for(var/datum/antagonist/separatist/separatist_datum in GLOB.antagonists)
+		var/independent_department_type = separatist_datum.owner?.assigned_role.departments_list[1]
+		independent_departments |= independent_department_type
+		team_datums |= separatist_datum.nation
+
+	if(!department)
+		//picks a random department if none was given
+		department = pick(list(/datum/job_department/assistant, /datum/job_department/medical, /datum/job_department/engineering, /datum/job_department/science, /datum/job_department/cargo, /datum/job_department/service, /datum/job_department/security) - independent_departments)
+		if(!department)
+			message_admins("Department Revolt could not create a nation, as all the departments are independent! You have created nations, you madman!")
+
+	department = SSjob.get_department_type(department)
+
+	for(var/datum/job/job as anything in department.department_jobs)
+		if(job.departments_list.len > 1 && job.departments_list[1] != department.type) //their loyalty is in other departments
+			continue
+		jobs_to_revolt += job.title
+
+	//setup team datum
+	var/datum/team/nation/nation = new(null, jobs_to_revolt, department)
+	nation.name = department.generate_nation_name()
+	var/datum/team/department_target //dodges unfortunate runtime
```

Comment:
I'm assuming you mean the pick from an empty list runtime, in which case write that

Reply:
carried over from the old one and i actually couldn't even remember which one haha so thank you